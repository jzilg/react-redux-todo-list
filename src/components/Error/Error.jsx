import React from 'react'
import PropTypes from 'prop-types'
import style from './error.scss'

const Error = ({ message }) => (
    <div className="container">
        <div className="error">
            <h2 className={style.title}>
                {message}
            </h2>
        </div>
    </div>
)

Error.propTypes = {
    message: PropTypes.string.isRequired,
}

export default Error
