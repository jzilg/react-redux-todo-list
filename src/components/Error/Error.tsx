import React, { ReactElement } from 'react'
import style from './error.scss'

interface ErrorProps {
    message: string,
}

const Error = ({ message }: ErrorProps): ReactElement<{}> => (
    <div className="container">
        <div className="error">
            <h2 className={style.title}>
                {message}
            </h2>
        </div>
    </div>
)

export default Error
