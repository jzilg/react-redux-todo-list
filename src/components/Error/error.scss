.link {
    color: #2e86c1;
}
