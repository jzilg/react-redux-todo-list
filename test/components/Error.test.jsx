import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'
import expect from 'expect'
import Error from '../../src/components/Error'

Enzyme.configure({ adapter: new Adapter() })

describe('Error', () => {
    it('should render correcty', () => {
        const message = 'An Error occurred'
        const component = shallow(<Error message={message} />)
        const tree = shallowToJson(component)
        expect(tree).toMatchSnapshot()
    })
})
