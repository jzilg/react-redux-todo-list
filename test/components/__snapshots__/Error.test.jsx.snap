// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render correcty 1`] = `
<div
  className="container"
>
  <div
    className="error"
  >
    <h2>
      An Error occurred
    </h2>
    <p>
      Is
       
      <a
        href="http://example.com"
        styleName="link"
      >
        http://example.com
      </a>
       
      active?
    </p>
  </div>
</div>
`;
